test/dynamo/test_ctx_manager.py

```diff
@@ -23,12 +28,31 @@ def __exit__(self, exc_type, exc_value, traceback):
         torch._C._set_grad_enabled(self.prev)
 
 
+@contextlib.contextmanager
+def customized_ctx_manager(mode):
+    prev = torch.is_grad_enabled()
+    try:
+        yield torch._C._set_grad_enabled(mode)
+    finally:
+        torch._C._set_grad_enabled(prev)
+
+
 class CustomizedCtxManagerWithGraphBreak(CustomizedCtxManager):
     def __enter__(self):
         torch._dynamo.graph_break()
         super().__enter__()
 
 
+@contextlib.contextmanager
```

Comment:
This is an interesting test case that demonstrates the InstructionTranslator parent problem:
```
def f(x):
    print(y)
    yield 1

g = f(1)

next(g) # this call should error out, because there's no y

g = f(1)

def h():
    y = 3
    next(g)

h()   # this call should succeed
```

The implication is that we need to change the parent to the new InstructionTranslator.

Reply:
The python model is we "suspend" the generator (on the yield) and continue executing the parent. The suspension keeps some state around (the frame state, which holds the locals). Some interesting reading: https://stackoverflow.com/questions/25232350/how-generators-work-in-python

We're going to need to figure out what "changing the parent" means. For example, we probably want to change the .parent field, and the child should inherit most of the values, like the parent's symbolic_globals and symbolic_torch_function_state. 

When we suspend the child (due to a yield call), we might need to merge the child's symbolic_globals back to the parent's symbolic_globals (if the child mutated the symbolic globals!). We may also need to do this with the "inconsistent_side_effects" (I don't know what these are).

I would recommend dumping a bunch of tests from https://github.com/python/cpython/blob/main/Lib/test/test_generators.py into the codebase.

Reply:
Some options:

1) `InliningGeneratorInstructionTranslator` sticks around and has "run()" and "suspend()" helper functions.
2) we create new InliningGeneratorInstructionTranslator every time we need to run a generator and we store exactly what the state of the generator is somewhere. The state of the generator is "the state of the python frame", which would be locals, the code, and the instruction pointer. Plus any "Dynamo-specific state" (for example, if TorchFunctionMode is active, Dynamo tracks that)